src/librustc_trans_utils/lib.rs

```diff
@@ -44,9 +47,62 @@ use rustc::hir;
 use rustc::hir::def_id::LOCAL_CRATE;
 use rustc::hir::map as hir_map;
 use rustc::util::nodemap::NodeSet;
+use rustc_mir::monomorphize::item::{MonoItem, MonoItemExt};
 
+pub mod assert_module_sources;
+pub mod diagnostics;
 pub mod link;
 pub mod trans_crate;
+pub mod symbol_names;
+pub mod symbol_names_test;
+
+#[inline(never)] // give this a place in the profiler
+pub fn assert_symbols_are_distinct<'a, 'tcx, I>(tcx: TyCtxt<'a, 'tcx, 'tcx>, trans_items: I)
```

Comment:
I'd rename `trans_` in this function to `mono_` and maybe move it to `librustc_mir/monomorphize`.

Reply:
Done locally